packages/core-react/src/context/reefKnot.tsx

```diff
@@ -24,13 +26,17 @@ export const ReefKnot = ({
   walletDataList,
   children,
 }: ReefKnotContextProps) => {
+  const [loadingWalletId, setLoadingWalletId] = useState<string | null>(null);
+
   const contextValue = useMemo(
     () => ({
       rpc,
       walletDataList,
       chains,
+      loadingWalletId,
+      setLoadingWalletId,
```

Comment:
instead of setter you can pass wrapper function to wrap connectAsync and do on/off state setting

Reply:
I don't see how it is more convenient. Some wallets don't need loaders and will use the standart connectAsync fn to connect, some will use this non-standart wrapper function

Reply:
Because you are repeating same try {set(true) } finally {set(false)}, and bound to make mistake somewhere